Objects/longobject.c

```diff
@@ -6773,3 +6774,110 @@ int PyLong_AsUInt64(PyObject *obj, uint64_t *value)
 {
     LONG_TO_UINT(obj, value, "C uint64_t");
 }
+
+
+static const PyLongLayout PyLong_LAYOUT = {
+    .bits_per_digit = PyLong_SHIFT,
+    .digits_order = -1,  // least significant first
+    .endian = PY_LITTLE_ENDIAN ? -1 : 1,
+    .digit_size = sizeof(digit),
+};
+
+
+const PyLongLayout* PyLong_GetNativeLayout(void)
+{
+    return &PyLong_LAYOUT;
+}
+
+
+int
+PyLong_Export(PyObject *obj, PyLongExport *export_long)
+{
+    if (!PyLong_Check(obj)) {
+        PyErr_Format(PyExc_TypeError, "expect int, got %T", obj);
+        return -1;
+    }
+
+    int64_t value;
+    int flags = Py_ASNATIVEBYTES_NATIVE_ENDIAN;
+    Py_ssize_t bytes = PyLong_AsNativeBytes(obj, &value, sizeof(value), flags);
+    if (bytes < 0) {
+        return -1;
```

Comment:
Hmm, that might fail.

I think it would be better to use the compact value right now.  It's (1) faster and (2) it doesn't fail.  Do you think that (2) is a severe restriction for future?

Reply:
> Hmm, that might fail.

In practice, it shouldn't fail. This error code is just because PyLong_AsNativeBytes() validates its parameters. But in this case, it cannot fail.

Reply:
I modified the code to be able to remove the error code path (`return -1;`, dead code).

Reply:
Yet another point is the speed.  I think that it's better to put in ``value`` either the compact value or the result of PyLong_AsLongAndOverflow.  Else - set up ``digits``.  I'll test this.

Reply:
On Linux, `long` and `int64_t` should be the same type. So you can use `mpz_set_si()`. Do you have concerns about Windows where `long` is only 32-bit?

For new APIs, I would prefer to use a type with a known size: int32_t, int64_t, etc. Avoid `long` type which has a different size depending on the platform: https://github.com/capi-workgroup/api-evolution/issues/10